sdk/test-utils/recorder/src/recorder.ts

```diff
@@ -123,14 +123,23 @@ export function record(testContext: Mocha.Context): Recorder {
         }
       }
     },
+
     getUniqueName: function(prefix: string, label?: string): string {
       let name: string;
       if (!label) {
         label = prefix;
       }
       if (isRecordMode()) {
         name = getUniqueName(prefix);
-        recorder.uniqueTestInfo["uniqueName"][label] = name;
+        if (recorder.uniqueTestInfo["uniqueName"][label]) {
```

Comment:
If we wanted to avoid any potential problem with weird labels e.g. `toString()` or `__proto__`, we could use a `Set` here instead of an object.

Reply:
weird labels? example?

Reply:
Thanks, leaving it as it is for now since it is of low priority and will take up a lot of efforts to revamp the recordings. Might revisit later.

Reply:
labels like "toString()".  Currently I believe we only use this for getting unique names for Azure resources and label is provided by users if they want to use a pre-existing prefix so we probably don't need to care for edge cases.

Reply:
Yup, discussed offline with @xirzec !